src/Neo/SmartContract/Native/NeoToken.cs

```diff
@@ -297,7 +304,16 @@ private void SetGasPerBlock(ApplicationEngine engine, BigInteger gasPerBlock)
         [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
         public BigInteger GetGasPerBlock(DataCache snapshot)
         {
-            return GetSortedGasRecords(snapshot, Ledger.CurrentIndex(snapshot) + 1).First().GasPerBlock;
+            var end = Ledger.CurrentIndex(snapshot) + 1;
+            var cached = snapshot.SerializedCache.Get<LastGasPerBlock>();
+            if (cached != null && cached.Index < end)
+            {
+                return cached.GasPerBlock;
+            }
+
+            var last = GetSortedGasRecords(snapshot, end).First();
```

Comment:
The main optimization is this one, avoiding the storage iteration for each call, but all "fees" are also affected